proto/v1/erc20.proto

```diff
@@ -2,63 +2,102 @@ syntax = "proto3";
 
 package erc20.types.v1;
 
-message BalanceChanges {
-  repeated BalanceChange balance_changes = 1;
+import "google/protobuf/timestamp.proto";
+
+message Events {
+  repeated Transfer transfers = 1;
+  repeated BalanceChange balance_changes = 2;
 }
 
 enum BalanceChangeType {
-  TYPE_UNKNOWN = 0; // cannot determine balance change
-  TYPE_1 = 1; // easy case where storage change is in the same call as the Transfer call
-  TYPE_2 = 2; // storage change is in a different call than the Transfer call
+  BALANCE_CHANGE_TYPE_UNSPECIFIED = 0; // cannot determine balance change
+  BALANCE_CHANGE_TYPE_1 = 1; // easy case where storage change is in the same call as the Transfer call
+  BALANCE_CHANGE_TYPE_2 = 2; // storage change is in a different call than the Transfer call
 }
 
 message BalanceChange {
-  string contract = 1;
-  string owner = 2;
-  string old_balance = 3;
-  string new_balance = 4;
+  // -- block --
+  uint64 block_num = 1;
+  string block_hash = 2;
+  google.protobuf.Timestamp timestamp = 3;
+  string date = 4;
 
-  string transaction = 5;
-  string storage_key = 6;
-  uint32 call_index = 7;
+  // -- transaction --
+  string transaction_id = 5;
 
-  string transfer_value = 8;
+  // -- call --
+  uint32 call_index = 6;
 
-  BalanceChangeType change_type = 9;
-}
+  // -- log --
+  uint32 log_index = 7; // Index is the index of the log relative to the transaction. This index is always populated regardless of the state revertion of the the call that emitted this log.
+  uint32 log_block_index = 8; // BlockIndex represents the index of the log relative to the Block.
+  uint64 log_ordinal = 9; // the block's global ordinal when the transfer was recorded.
 
-message ValidBalanceChanges {
-  repeated ValidBalanceChange valid_balance_changes = 1;
-}
+  // -- storage change --
+  string storage_key = 10;
+  uint64 storage_ordinal = 11;
```

Comment:
See comment about ordinals on `version` field in this file. We should remove this one or maybe replace by our "relative" ordering. But the field `version` already acts as that, so outside of debugging, we should I think not add this.